pkg/rbd/rbd_util.go

```diff
@@ -813,3 +812,59 @@ func cleanupRBDImageMetadataStash(path string) error {
 
 	return nil
 }
+
+func flattenRbdImage(ctx context.Context, rbdVol *rbdVolume, maxDepth uint, cr *util.Credentials) error {
+
+	depth, err := getCloneDepth(ctx, rbdVol.Monitors, rbdVol.Pool, rbdVol.RbdImageName, 0, cr)
+	if err != nil {
+		return err
+	}
+
+	klog.Infof(util.Log(ctx, "image depth is %v and configured clone depth hard limit is %v"), depth, maxDepth)
+	if depth >= int(maxDepth) {
+		klog.Infof(util.Log(ctx, "maximum clone depth (%d) has been reached, flattening %v volume "), maxDepth, rbdVol.RbdImageName)
+		args := []string{"-m", rbdVol.Monitors,
+			"--id", cr.ID,
+			"--pool", rbdVol.Pool,
+			"--no-progress",
+			"--image", rbdVol.RbdImageName,
+			"--keyfile=" + cr.KeyFile,
+			"-c", util.CephConfigPath,
+			"flatten"}
```

Comment:
We should still be using `ceph rbd task add flatten` if possible

Reply:
can we add this support once we add a soft limit?

Reply:
We really don't want to be blocking the CSI -- that's why we added the background task stuff to begin with.

Reply:
... I think this comment is still valid.